packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -5,55 +5,90 @@
 
 import * as vscode from 'vscode'
 import * as fs from 'fs'
+import * as path from 'path'
 import { getLogger, Logger } from '../../shared/logger'
 import { localize } from '../../shared/utilities/vsCodeUtils'
 import { VueWebview } from '../../webviews/main'
 import { ExtContext } from '../../shared/extensions'
-//import { telemetry } from '../../shared/telemetry/telemetry'
+// TODO: Implement Telemetry ... import { telemetry } from '../../shared/telemetry/telemetry'
 import { AccessAnalyzer, S3 } from 'aws-sdk'
-import { exec } from 'child_process'
+import { execSync } from 'child_process'
+import { ToolkitError } from '../../shared/errors'
 const AmazonS3URI = require('amazon-s3-uri')
 
+const defaultTerraformConfigPath = path.join(__dirname, '..', 'terraformConfig', 'default.yaml')
+const tempFilePath = path.join(__dirname, 'policyDocument.json')
+const customPolicyCheckDiagnosticCollection = vscode.languages.createDiagnosticCollection('customPolicyCheck')
+const validatePolicyDiagnosticCollection = vscode.languages.createDiagnosticCollection('validatePolicy')
+
 export interface IamPolicyChecksInitialData {
     referenceFilePath: string
-    cfnParameterPath: string
     referenceDocument: string
+    referenceFileErrorMessage: string
+    cfnParameterPath: string
     pythonToolsInstalled: boolean
 }
 
 export class IamPolicyChecksWebview extends VueWebview {
     public static readonly sourcePath: string = 'src/accessanalyzer/vue/index.js'
     public readonly id = 'iamPolicyChecks'
-
-    //private readonly logger = getLogger()
+    private static editedDocumentUri: vscode.Uri
+    private static editedDocumentFileName: string
+    private static editedDocument: string
 
     public constructor(
         private readonly data: IamPolicyChecksInitialData,
         private readonly client: AccessAnalyzer,
+        private readonly region: string,
         public readonly onChangeInputPath = new vscode.EventEmitter<string>(),
         public readonly onChangeReferenceFilePath = new vscode.EventEmitter<string>(),
-        public readonly onChangeCloudformationParameterFilePath = new vscode.EventEmitter<string>()
+        public readonly onChangeCloudformationParameterFilePath = new vscode.EventEmitter<string>(),
+        public readonly onValidatePolicyResponse = new vscode.EventEmitter<[string, string]>(),
+        public readonly onCustomPolicyCheckResponse = new vscode.EventEmitter<[string, string]>(),
```

Comment:
Q: guess it goes back to design - does this mean customers click on different button or use different views for linter/treehouse checks?

Reply:
There are two sections, one for ValidatePolicy and one for CustomChecks